// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTNym";
option optimize_for = LITE_RUNTIME;

import public "Authority.proto";
import public "Enums.proto";
import public "NymIDSource.proto";

message Nym
{
    optional uint32 version = 1;
    optional string nymID = 2;
    optional NymMode mode = 3;
    optional uint32 index = 4;
    optional uint64 revision = 5;
    optional NymIDSource source = 6;
    repeated Authority activeCredentials = 7;
    repeated Authority revokedCredentials = 8;
}
